SourceDistributions/Developer-Sandboxes/tsWxGTUI_PyVx/Python-3x/tsWxGTUI_Py3x/ManPages/tsManPagesLibGUI/tsWxEventQueueEntry.man

Help on module tsWxEventQueueEntry:

NAME
    tsWxEventQueueEntry

FILE
    /cygdrive/d/WR/SoftwareGadgetry-PyPI/Python-2x/tsWxGTUI_Py2x/tsManPagesLibGUI/tsWxEventQueueEntry.py

DESCRIPTION
    tsWxEventQueueEntry.py - Class to establish a container for
    the triggering object, triggering data and triggering event.

CLASSES
    __builtin__.object
        EventQueueEntry
    
    class EventQueueEntry(__builtin__.object)
     |  Establish a  a container for the triggering object, triggering data and
     |  triggering event.
     |  
     |  Methods defined here:
     |  
     |  GetTriggeringData(self)
     |      Return TriggeringData
     |  
     |  GetTriggeringEvent(self)
     |      Return TriggeringEvent
     |  
     |  GetTriggeringObject(self)
     |      Return TriggeringObject
     |  
     |  GetTriggeringObjectParent(self)
     |      Return Parent of TriggeringObject
     |  
     |  __del__(self)
     |      Destroy a container for triggering object and triggering event.
     |  
     |  __init__(self, triggeringObject=None, triggeringEvent=None, triggeringData=None)
     |      Construct a container for triggering object and triggering event.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors defined here:
     |  
     |  TriggeringData
     |      Return TriggeringData
     |  
     |  TriggeringEvent
     |      Return TriggeringEvent
     |  
     |  TriggeringObject
     |      Return TriggeringObject
     |  
     |  TriggeringObjectParent
     |      Return Parent of TriggeringObject
     |  
     |  __dict__
     |      dictionary for instance variables (if defined)
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)

DATA
    DEBUG = True
    VERBOSE = True
    __authors__ = 'Richard S. Gordon'
    __copyright__ = 'Copyright (c) 2007-2013 Richard S. Gordon.\n\t\tAll r...
    __credits__ = '\n\n  Credits: \n\n\t  tsLibGUI Import & Application .....
    __date__ = '04/01/2013'
    __header__ = '\n\ntsWxEventQueueEntry, v1.1.0 (build 04/01/2013)... Ge...
    __license__ = 'GNU General Public License, Version 3, 29 June 2007'
    __line1__ = 'tsWxEventQueueEntry, v1.1.0 (build 04/01/2013)'
    __line2__ = 'Author(s): Richard S. Gordon'
    __line3__ = 'Copyright (c) 2007-2013 Richard S. Gordon.\n\t\tAll right...
    __line4__ = 'GNU General Public License, Version 3, 29 June 2...U Gene...
    __title__ = 'tsWxEventQueueEntry'
    __version__ = '1.1.0'
    mainTitleVersionDate = 'tsWxEventQueueEntry, v1.1.0 (build 04/01/2013)...

VERSION
    1.1.0

DATE
    04/01/2013

CREDITS
    
    
      Credits: 
    
    	  tsLibGUI Import & Application Launch Features: 
    	  Copyright (c) 2007-2009 Frederick A. Kier.
    			All rights reserved.
    
    	  Python Curses Module API & Run Time Library Features:
    	  Copyright (c) 2001-2013 Python Software Foundation.
    			All rights reserved.
    	  PSF License Agreement for Python 2.7.3 & 3.3.0
    
    	  wxWidgets (formerly wxWindows) & wxPython API Features:
    	  Copyright (c) 1992-2008 Julian Smart, Robert Roebling,
    			Vadim Zeitlin and other members of the 
    			wxWidgets team.
    			All rights reserved.
    	  wxWindows Library License
    
    	  nCurses API & Run Time Library Features:
    	  Copyright (c) 1998-2011 Free Software Foundation, Inc.
    			All rights reserved.
    	  GNU General Public License, Version 3, 29 June 2007




Documents/NEWS.txt
#-----------------------------------------------------------
#"Time-stamp: <07/08/2015  4:48:41 AM rsg>"
#-----------------------------------------------------------

========================= NEWS.txt =========================

   +----+----+  TeamSTARS "tsWxGTUI_PyVx" Toolkit
   | ts | Wx |      with Python 2x & Python 3x based
   +----+----+         Command Line Interface (CLI)
   | G T U I |      and "Curses"-based "wxPython"-style, 
   +---------+         Graphical-Text User Interface (GUI)
 
   Get that cross-platform, pixel-mode "wxPython" feeling
   on character-mode 8-/16-color (xterm-family) & non-color
   (vt100-family) terminals and terminal emulators.

   You can find this and other plain-text files in the
   Toolkit subdirectory named:

       "./<Toolkit Recipient's Repository>/Documents".

       <Your Working Repository>
       (e.g. "tsWxGTUI_PyVx_Repository") 
         |
         |
         +-- ["Documents"]

=========================== News ===========================

The pre-alpha release of the TeamSTARS "tsWxGTUI_PyVx-0.0.0"
Toolkit is now available. It includes the following two
versions abd their associated variants (Python Distribution
Utilities ("Distutils") installable "Site-Packages" and non-
installable "Developer-Sandboxes" for enhancement and bug
fix development and experimentation without interference
with the already installed Site-Packages):

1. The TeamSTARS "tsWxGTUI_Py2x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   second generation Python 2.x programming language which
   is widely available.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. The final 2.x
   version 2.7 release came out in mid-2010, with a state-
   ment of extended support for this end-of-life release.
   The 2.x branch will see no new major releases after that.

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 2.0.0-2.7.9.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 2.6.4-2.7.9.

2. The TeamSTARS "tsWxGTUI_Py3x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   third generation Python 3.x programming language which
   is growing in availability.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. 3.x is under
   active development and has already seen over five years
   of stable releases, including version 3.3 in 2012 and
   3.4 in 2014. This means that all recent standard library
   improvements, for example, are only available by default
   in Python 3.x.

   Guido van Rossum (the original creator of the Python
   language) decided to clean up Python 2.x properly,
   with less regard for backwards compatibility than is
   the case for new releases in the 2.x range. The most
   drastic improvement is the better Unicode support
   (with all text strings being Unicode by default) as
   well as saner bytes/Unicode separation.

   Besides, several aspects of the core language (such as
   print and exec being statements, integers using floor
   division) have been adjusted to be easier for newcomers
   to learn and to be more consistent with the rest of the
   language, and old cruft has been removed (for example,
   all classes are now new-style, "range()" returns a memory
   efficient iterable, not a list as in 2.x).

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 3.0.0-3.4.2.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 3.1.5-3.4.2.

======================= End-Of-File ========================


SourceDistributions/Developer-Sandboxes/tsWxGTUI_PyVx/Python-3x/tsWxGTUI_Py3x/tsLibGUI/tsWxPkg/src/tsWxEventQueueEntry.py
#! /usr/bin/env python
# "Time-stamp: <10/25/2013  7:31:14 AM rsg>"
'''
tsWxEventQueueEntry.py - Class to establish a container for
the triggering object, triggering data and triggering event.
'''
#################################################################
#
# File: tsWxEventQueueEntry.py
#
# Purpose:
#
#    Class to establish a container for the triggering object,
#    triggering data and triggering event.
#
# Usage (example):
#
#    # Import
#
#    from tsWxEventQueueEntry import EventQueueEntry
#
# Requirements:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#    See \WR\wxWidgets-2.8.10\src\common\event.cpp
#
# Capabilities:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Limitations:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Notes:
#
#    None.
#
# Methods:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Classes:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Modifications:
#
#    2012/05/21 rsg Added support for triggeringData that
#                   conveys instance specific information from
#                   the triggeringObject that is associacted
#                   with the triggeringEvent.
#
#    2012/05/21 rsg Re-organized the methods into alphabetical order.
#
# ToDo:
#
#    None.
#
#################################################################

__title__     = 'tsWxEventQueueEntry'
__version__   = '1.1.0'
__date__      = '04/01/2013'
__authors__   = 'Richard S. Gordon'
__copyright__ = 'Copyright (c) 2007-2013 ' + \
                '%s.\n\t\tAll rights reserved.' % __authors__
__license__   = 'GNU General Public License, ' + \
                'Version 3, 29 June 2007'
__credits__   = '\n\n  Credits: ' + \
                '\n\n\t  tsLibGUI Import & Application Launch Features: ' + \
                '\n\t  Copyright (c) 2007-2009 Frederick A. Kier.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\n\t  Python Curses Module API & ' + \
                'Run Time Library Features:' + \
                '\n\t  Copyright (c) 2001-2013 ' +\
                'Python Software Foundation.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  PSF License Agreement for Python 2.7.3 & 3.3.0' + \
                '\n\n\t  wxWidgets (formerly wxWindows) & ' + \
                'wxPython API Features:' + \
                '\n\t  Copyright (c) 1992-2008 Julian Smart, ' + \
                'Robert Roebling,' + \
                '\n\t\t\tVadim Zeitlin and other members of the ' + \
                '\n\t\t\twxWidgets team.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  wxWindows Library License' + \
                '\n\n\t  nCurses API & Run Time Library Features:' + \
                '\n\t  Copyright (c) 1998-2011 ' + \
                'Free Software Foundation, Inc.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  GNU General Public License, ' + \
                'Version 3, 29 June 2007'

__line1__ = '%s, v%s (build %s)' % (__title__, __version__, __date__)
__line2__ = 'Author(s): %s' % __authors__
__line3__ = '%s' % __copyright__

if len( __credits__) == 0:
    __line4__ = '%s' % __license__
else:
    __line4__ = '%s%s' % (__license__, __credits__)

__header__ = '\n\n%s\n\n  %s\n  %s\n  %s\n' % (__line1__,
                                               __line2__,
                                               __line3__,
                                               __line4__)

mainTitleVersionDate = __line1__

##import tsWxGlobals as wx # Invalid import because of circular dependency.
import tsCxGlobals as cx

#---------------------------------------------------------------------------

##DEBUG = True # Avoids tsWx import because of circular dependency.
DEBUG = cx.ThemeToUse['Troubleshooting']['Debug_GUI_Launch'] | \
        cx.ThemeToUse['Troubleshooting']['Debug_GUI_Progress'] | \
        cx.ThemeToUse['Troubleshooting']['Debug_GUI_Termination'] | \
        cx.ThemeToUse['Troubleshooting']['Debug_GUI_Exceptions']

##VERBOSE = True # Avoids tsWx import because of circular dependency.
VERBOSE = cx.ThemeToUse['Troubleshooting']['Debug_GUI_Configuration']

#---------------------------------------------------------------------------

class EventQueueEntry(object):
    '''
    Establish a  a container for the triggering object, triggering data and
    triggering event.
    '''
    def __init__(self,
                 triggeringObject=None,
                 triggeringEvent=None,
                 triggeringData=None):
        '''
        Construct a container for triggering object and triggering event.
        '''
        self.ts_TriggeringObject = triggeringObject
        self.ts_TriggeringEvent = triggeringEvent
        self.ts_TriggeringData = triggeringData

    #-----------------------------------------------------------------------
 
    def __del__(self):
        '''
        Destroy a container for triggering object and triggering event.
        '''
        del self.ts_TriggeringObject
        del self.ts_TriggeringEvent
        del self.ts_TriggeringData
        del self

    #-----------------------------------------------------------------------

    def GetTriggeringData(self):
        '''
        Return TriggeringData
        '''
        return (self.ts_TriggeringData)

    #-----------------------------------------------------------------------

    def GetTriggeringEvent(self):
        '''
        Return TriggeringEvent
        '''
        return (self.ts_TriggeringEvent)

    #-----------------------------------------------------------------------
 
    def GetTriggeringObject(self):
        '''
        Return TriggeringObject
        '''
        return (self.ts_TriggeringObject)

    #-----------------------------------------------------------------------
 
    def GetTriggeringObjectParent(self):
        '''
        Return Parent of TriggeringObject
        '''
        try:
            return (self.ts_TriggeringObject.ts_Parent)
        except Exception as e:
            return (None)

    #-----------------------------------------------------------------------

    TriggeringData = property(GetTriggeringData)
    TriggeringEvent = property(GetTriggeringEvent)
    TriggeringObject = property(GetTriggeringObject)
    TriggeringObjectParent = property(GetTriggeringObjectParent)

#---------------------------------------------------------------------------

if __name__ == '__main__':

    print(__header__)
